BlueprintUI/Sources/Element/ElementContent.swift

```diff
@@ -45,8 +45,19 @@ public struct ElementContent {
         cache: CacheTree,
         layoutMode: LayoutMode
     ) -> CGSize {
-        // TODO: switch on layoutMode
-        storage.measure(in: constraint, environment: environment, cache: cache)
+        switch layoutMode {
+        case .standard:
+            return storage.measure(in: constraint, environment: environment, cache: cache)
+        case .singlePass:
+            return storage.sizeThatFits(
+                proposal: ProposedViewSize(constraint),
+                context: .init(
+                    // TODO: Hoist upward?
```

Comment:
Should we do this before we get too deep into performance optimizations? Might this have a significant impact?

Reply:
Yeah, I'll make a note in our planning doc